clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -547,6 +547,14 @@ CVarSet
         T = {P};
 
         Ret = T;
+      } else if (StmtExpr *SE = dyn_cast<StmtExpr>(E)) {
+        CVarSet T;
+        // retrieve the last "thing" returned by the block
+        Stmt *Res = SE->getSubStmt()->getStmtExprResult();
+        if(Expr *ESE = dyn_cast<Expr>(Res)) {
+          ESE = ESE->IgnoreParens();
```

Comment:
I am pretty sure this line is unnecessary. The start of `getExprConstraintVars` will remove the parens anyway.
```
CVarSet
    ConstraintResolver::getExprConstraintVars(Expr *E) {
  CVarSet EmptyCSet;
  if (E != nullptr) {
    auto &CS = Info.getConstraints();
    QualType TypE = E->getType();
    E = E->IgnoreParens();
```